hadoop-crypto/src/main/java/com/palantir/hadoop/EncryptedFileSystem.java

```diff
@@ -76,6 +79,29 @@ public FSDataOutputStream create(Path path, FsPermission permission,
         return os;
     }
 
+    @Override
+    public boolean rename(Path src, Path dst) throws IOException {
+        KeyMaterial keyMaterial = keyStore.get(src.toString());
+        keyStore.put(dst.toString(), keyMaterial);
```

Comment:
leave comment explaining the convoluted code path (provides safety net in case `put` fails)